src/content/post/20161122_drupal_application_bottlenecks_avoid_with_profiling.md

```diff
@@ -0,0 +1,160 @@
++++
+date = "2016-11-22T11:02:30Z"
+draft = false
+title = "Avoid bottlenecks on a Drupal web application and how identify it with profiling"
+description = "How identify bottleneck of your web application and avoid that can appear in the future"
+tags        = [ "profiling", "drupal", "performance", "web application"  ]
+topics      = [ "Profiling", "Drupal", "Performance" ]
+author = "Vinz"
++++
+
+### TL;DR
+
+This article is focused on the possible *bottlenecks of a web application*, how you can identify it and improve the performance of your code.
+
+We'll talk about using *Blackfire* to get insights on a Drupal-based PHP application running on a LAMP stack.
+
+<!-- more -->
+
+Application's performance is one of the feature that tells a *done* job from a *good* job, and shows a focus on quality.
+
+A customer's call about website performances, given server infrastructure is adequate to requests, most probably indicates some errors in our code.
+
+Googling for 'web application performance checklist' can surely point out some good general solutions to check out first, but if problems persist we have some specific issue we need to identify and solve.
```

Comment:
Da qui in giù toglierei tutto e scriverei invece una cosa simile (la metto qui in Italiano poi facciamo review in inglese così fai esercizio ;-) ):

```
Lo scopo del post è di fornire una quick-reference da poter consultare nel momento in cui un progetto Drupal va in produzione.
Ci sono due tipologie di operazioni per migliorare le performance: l'applicazione di best-practice, valide a prescindere dalle funzioni del prodotto, e il profiling della logica, più dipendente dal dominio e dall'implementazione e che può essere considerata la ciliegina sulla torta.

Forniremo quindi:

* Un elenco di controlli, validi per una qualsiasi applicazione PHP, con indicazioni su quali componenti di Drupal sono coinvolti punto per punto.
* Strumenti e metodi per ottenere informazioni su come la nostra applicazione sta performando e dove sono i colli di bottiglia.
* Un esempio pratico di come trovare i punti di miglioramento e applicare i dovuti cambiamenti.
* Qualche pro-tip dettati dall'esperienza sulle trappole [pitfall, non traps ndP] più comuni e come aggirarli.
```

Più sotto vedi i suggerimenti su come cambiare la struttura dei titoli per mimare questo elenco, in modo che uno si trova con quanto c'è scritto. :)